docs/src/manual/nlp.md

```diff
@@ -435,6 +431,40 @@ register(model, :my_square, 1, f, ∇f, ∇²f)
 @NLobjective(model, Min, my_square(x))
 ```
 
+#### Multivariate functions
+
+For multivariate functions, the hessian function `∇²f` must take an
+`AbstractMatrix` as the first argument, the lower-triangular of which is filled
+in-place:
+```@example
+using JuMP #hide
+f(x...) = (1 - x[1])^2 + 100 * (x[2] - x[1]^2)^2
+function ∇f(g, x...)
+    g[1] = 400 * x[1]^3 - 400 * x[1] * x[2] + 2 * x[1] - 2
+    g[2] = 200 * (x[2] - x[1]^2)
+    return
+end
+function ∇²f(H, x...)
+    H[1, 1] = 1200 * x[1]^2 - 400 * x[2] + 2
+    # H[1, 2] = -400 * x[1]  <-- Not needed. Fill the lower-triangular only.
+    H[2, 1] = -400 * x[1]
+    H[2, 2] = 200.0
+    return
+end
+
+model = Model()
+register(model, :rosenbrock, 2, f, ∇f, ∇²f)
+@variable(model, x[1:2])
+@NLobjective(model, Min, rosenbrock(x[1], x[2]))
+```
+
+!!! warning
+    Make sure the first argument to `∇²f` supports an `AbstractMatrix`, and do
+    not assume the input is `Float64`. You may assume the matrix is initialized
+    with zeros, so you need only to fill in the non-zero terms. The matrix type
+    passed in as `H` is very limited. You may assume only that it supports
```

Comment:
Suggestion: rephrase 
```
The matrix type passed in as `H` is very limited.
```
with 
```
By default, the Hessian function supports a broad range of `AbstractMatrix` format, as long as it supports `size(H)` and `setindex!`.
```

Reply:
I changed the wording slightly, it's not that the Hessian function supports a broad range, it's that the AD system passes in something that acts like a matrix but isn't really, so user functions are limited in what they can do.